src/services/apps/nats/main.py

```
@@ -0,0 +1,327 @@
+    leader_node_id = get_leader_node(state_json)
+    is_leader = (leader_node_id == local_node_id)
+    cluster_initialized = is_cluster_initialized(nats_props)
+
+    local_tunnel_ip = get_node_tunnel_ip(local_node_id, wg_props)
+    if not local_tunnel_ip:
+        return PluginOutput(status="error", error_message="Local node has no WireGuard tunnel IP", local_state=local_state)
+
+    # Write NATS config based on current cluster view
+    try:
+        write_nats_config(local_node_id, local_tunnel_ip, cluster_nodes, wg_props)
+    except Exception as e:
```

Ruff (0.14.11)

Do not catch blind exception: `Exception`

(BLE001)